taurus/demo/cake.py

```diff
@@ -154,6 +159,9 @@ def make_cake_templates():
     tmpl["Procurement"] = ProcessTemplate(name="Procurement",
                                           description="Buyin' stuff")
 
+    for key in tmpl:
+        tmpl[key].add_uid(DEMO_SCOPE, key)
+
```

Comment:
This can be removed. It's true that it is slightly different from the id assignment below, in that it uses the key in the template dictionary, which isn't necessarily equal to the name ("Generic Material" and "Baked Color", for example). But the ids assigned here are overwritten by the recursive_foreach, since they all use the same scope.